### PR TITLE
Make genOpcodeFiles.ml handle opcode arity.

### PR DESCRIPTION
**Pull request by silene**
The handwritten table in `coq_fix_code.c` has been an endless source of critical bugs, and sometimes they even reached `master` without being detected, e.g., #13867.

To avoid introducing brand new bugs in the process, the first commit of the PR does compare the generated file with the hand-written one. The second then removes the hand-written code. Finally, the third commit makes the code a bit less permissive.

This depends on #13867 to avoid generating an arity table known to be broken.

### CODE REVIEW

**Comment by gares** [comment]
In the patch you say it's fixed in 8.13.1. If so, change the milestone.

**Comment by silene** [comment]
> In the patch you say it's fixed in 8.13.1. If so, change the milestone.

It is #13867 which targets 8.13.1. This one does not (unless you really want it).

**Comment by MSoegtropIMC** [comment]
I restarted the GitLab Windows CI test. I don't know much about the GitHub/Azure test. Afaik it also uses cygwin + opam which means one has to wait until the opam fix is merged and propagated to the MinGW opam repo, which will take a few days (Coq Platform has a local opam patch repo which allows for temporary preview fixes).

**Review by gares — APPROVED** [review]
build systems seem ok to me

**Review by SkySkimmer — APPROVED** [review]
OK for me

**Comment by SkySkimmer** [comment]
@ppedrot you are the only vm-maintainer who is `not (author || away from coq work)`, can you take this?

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
Merging soon.

**Pull request merged**
